nextcloudappstore/core/models.py

```diff
@@ -512,3 +512,62 @@ class Meta:
     def __str__(self) -> str:
         return '%s: %s %s' % (self.app_release.app, self.php_extension,
                               self.version_spec)
+
+
+class AppOwnershipTransfer(Model):
+    """Represents a transfer of ownership of an app from one user to another.
+
+    This model fulfills two purposes:
+
+    - Be a proposal of an app ownership transfer that may or may not be
+    accepted by the involved users.
+    - Perform (commit) the transfer.
+
+    When a transfer object is created, the field 'old_owner' is automatically
+    set to the owner of the 'app'. Thus, to initiate a transfer, use the
+    following statement:
+
+        AppOwnershipTransfer.objects.create(app=app, new_owner=user)
+
+    An instance of AppOwnershipTransfer is deleted when the transfer it
+    represents is committed.
+    """
+
+    app = OneToOneField(
+        App, on_delete=CASCADE, related_name='ownership_transfer')
+    old_owner = ForeignKey(
+        User,
+        related_name='app_ownership_transfers_outgoing',
+        on_delete=CASCADE,
+        null=False)
+    new_owner = ForeignKey(
+        User,
+        related_name='app_ownership_transfers_incoming',
+        on_delete=CASCADE,
+        null=False)
+    is_accepted_old_owner = BooleanField(default=False)
```

Comment:
Probably irrelevant to store if both users accepted it. If a request is created we will render it. If its approved we will delete the entry and change the owner on the app